apps/messagegui/app-newmessage.js

```diff
@@ -3,3 +3,18 @@ BUZZ_ON_NEW_MESSAGE is set so when messagegui.app.js loads it knows
 that it should buzz */
 global.BUZZ_ON_NEW_MESSAGE = true;
 eval(require("Storage").read("messagegui.app.js"));
+
+// Add temporary listeners for unlocking watch. `settings` is init:ed in the eval:ed file "messagegui.app.js".
+if (settings.tempWakeOnFaceUp || settings.tempWakeOnTwist) {
```

Comment:
@atjn This gives the lint warning:
` Warning:    8:5   warning  'settings' is not defined  no-undef`

It's defined correctly in the evaluated file.

Should we add an exemption for this maybe?

Reply:
The only real solution to this, is to not use `eval`. `eval` has many downsides, one of them being that it is impossible to analyze it with tools like Eslint. Preferably, Espruino should support `require("messagegui.app.js").settings` or something like it, because then it is clear where the variable is coming from. Unfortunately I cannot find a simple solution in Espruino that would support this pattern without using `eval`. Maybe you have some ideas?

The quick and dirty solution is to define `settings` as a global variable. You already have a comment about this to help human readers understand the code, you just need to format it so that Eslint understands it as well. See my suggestion for how to format it correctly :)

Reply:
I realized that it can be initiated as undefined first in this filed and then be defined in the evaluation. Like with the latest commit. That avoids the lint warning. I think we can go with this - what do you think?

Reply:
Yes that seems like a fine solution as well, but beware that if the declaration in the source file is changed to a `let` or `const`, it will break the `eval`, and unless you are testing the app thoroughly, you might not be aware that you are breaking something.